controllers/notes.controller.js

```diff
@@ -0,0 +1,94 @@
+const { Note } = require("../models/note.model");
+const { extend } = require("lodash");
+
+const getNotesByUserIdAndVideoId = async (req, res) => {
+  try {
+    const userId = req.get("userId");
+    const videoId = req.get("videoId");
+    const notesFromDb = await Note.find({ userId: userId, videoId: videoId });
+    res.status(200).json({ success: true, notes: notesFromDb });
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      success: false,
+      message: "Something went wrong!",
+      errMessage: err.message,
+    });
+  }
+};
+
+const addNewNote = async (req, res) => {
+  try {
+    const noteData = req.body;
+    const NewNote = new Note(noteData);
+    const savedNote = await NewNote.save();
+    res.status(201).json({ success: true, note: savedNote });
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      success: false,
+      message: "Something went wrong!",
+      errMessage: err.message,
+    });
+  }
+};
+
+const getNoteByIdFromDb = async (req, res, next, id) => {
+  try {
+    const noteFromDb = await Note.findById(id);
+    if (!noteFromDb) {
+      return res
+        .status(404)
+        .json({ success: false, message: "Note not found" });
+    }
+    req.noteFromDb = noteFromDb;
+    next();
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      success: false,
+      message: "Something went wrong!",
+      errMessage: err.message,
+    });
+  }
+};
+
+const updateNote = async (req, res) => {
+  try {
+    let { noteFromDb } = req;
```

Comment:
Are you trying to destructure noteFromDb from req.body here?

Reply:
No that is from a middleware to add the noteFromDb to the req object